### DISCUSSION TITLE
Lifecycle functions in common part in KMP

### DISCUSSION THREAD

**Discussion by ahmadhassan5** [Tlaster/PreCompose#166] [Q&A] [answered]
Hi!
How can we use ON_PAUSE and ON_RESUME lifecycle functions in Compose Multiplatform as in Jetpack Compose ?
PreCompose provide any way?

**Comment by Tlaster** [comment, marked as answer]
Yes, there's also a `moe.tlaster.precompose.lifecycle.LocalLifecycleOwner`, you can use this to handle lifecycle event, there're only 4 lifecycle states in PreCompose: Initialized, Active, InActive, Destroyed.

**Reply by ahmadhassan5** [reply]
Thank you for the response.

I think you introduced it recently, it was not present before.
InActive is like ON_STOP?

And when Destroyed is called?
If I navigate to another screen, no event is being triggered.

**Reply by Tlaster** [reply]
On Android [InActive is mapping to onPause globally](https://github.com/Tlaster/PreCompose/blob/ce960fcc4cceb5288dc8c1b593ac583a9a2e8183/precompose/src/androidMain/kotlin/moe/tlaster/precompose/lifecycle/PreComposeActivity.kt#L99), and Destroyed is being called when a route [is removed from back stack](https://github.com/Tlaster/PreCompose/blob/ce960fcc4cceb5288dc8c1b593ac583a9a2e8183/precompose/src/commonMain/kotlin/moe/tlaster/precompose/navigation/BackStackManager.kt#L167).
If you want to handle lifecycle event per screen, you need to call `moe.tlaster.precompose.lifecycle.LocalLifecycleOwner` inside your screen.

**Reply by ahmadhassan5** [reply]
<img width="820" alt="image" src="https://github.com/Tlaster/PreCompose/assets/63135934/44d1199f-9aeb-45b0-8f96-8a47a4ffd401">


Yes, using LocalLifecycleOwner. Check the attached screenshot.
Let me know if changes needed

**Reply by Tlaster** [reply]
In `onDispose` you remove the observer, and `onDispose` is being called when composable is removed from composable tree, which can happens before precompose set the lifecycle to InActive state.
If all you need is just save something when navigating to another screen, you can just use `onDispose` without any lifecycle.

**Reply by ahmadhassan5** [reply]
Actually I was assuming its behaviour like Android native side. 

Anyway, thank you for your time and amazing library.
